src/test/kotlin/com/exactpro/th2/common/message/TestMessageFilterUtils.kt

```diff
@@ -84,6 +103,32 @@ class TestMessageFilterUtils {
         Assertions.assertEquals(rootMessageFilterJson, objectMapper.writeValueAsString(toTreeTable))
     }
 
+    @ParameterizedTest
+    @MethodSource("additionalMetadata")
+    fun `valid root message filter to readable body collection conversion`(additionalMetadata: Map<String, String>) {
+        val expectedJson =
+            readableRootMessageFilterJson.replace(ADDITIONAL_METADATA_TAG, mapToJsonConverter(additionalMetadata))
+        val toTreeTable = RootMessageFilter.newBuilder().apply {
+            messageType = "MsgType"
+            messageFilter = createMessageFilter()
+            metadataFilterBuilder.apply {
+                putPropertyFilters("propA", simplePropertyFilter("valA", NOT_EQUAL, true))
+                putPropertyFilters("propB", simplePropertyFilter("valB"))
+            }
+            comparisonSettingsBuilder.apply {
+                addIgnoreFields("fieldA")
+                addIgnoreFields("fieldB")
+            }
+        }.build().toReadableBodyCollection(additionalMetadata)
+
+        val expected = objectMapper.readTree(expectedJson).toList()
+        val actual = objectMapper.valueToTree<JsonNode>(toTreeTable).toList()
+
+        Assertions.assertNotNull(toTreeTable)
```

Comment:
I think we should check it before calling `objectMapper.valueToTree` on line 125